src/vectara/client.py

```diff
@@ -33,5 +97,83 @@ def set_upload_manager(self, upload_manager: UploadManager) -> None:
     def set_lab_helper(self, lab_helper: LabHelper) -> None:
         self.lab_helper = lab_helper
 
-class AsyncVectara(AsyncBaseVectara): 
-    pass
\ No newline at end of file
+    def chat(
+            self,
+            *,
+            query: str,
+            search: SearchCorporaParameters = OMIT,
+            request_timeout: typing.Optional[int] = None,
+            request_timeout_millis: typing.Optional[int] = None,
+            generation: typing.Optional[GenerationParameters] = OMIT,
+            chat: typing.Optional[ChatParameters] = OMIT,
+            chat_id: typing.Optional[str] = None,
+            request_options: typing.Optional[RequestOptions] = None,
+    ) -> ChatFullResponse:
+        """
+        Modified `chat` method to support chat_id for session reuse.
+
+        Parameters
+        ----------
+        query : str
+            The chat message or question.
+        search : typing.Optional[SearchCorporaParameters]
+            Retrieval parameters. Optional if chat_id is provided.
+        request_timeout : typing.Optional[int]
+            The API will make a best effort to complete the request in the specified seconds or time out.
+        request_timeout_millis : typing.Optional[int]
+            The API will make a best effort to complete the request in the specified milliseconds or time out.
+        generation : typing.Optional[GenerationParameters]
+            Parameters for response generation.
+        chat : typing.Optional[ChatParameters]
+            Chat-specific parameters. Optional if chat_id is provided.
+        chat_id : typing.Optional[str]
+            ID of the chat session to reuse stored session data.
+        request_options : typing.Optional[RequestOptions]
+            Request-specific configuration.
+
+        Returns
+        -------
+        ChatFullResponse
+        """
+
+        if not chat_id and not search:
+            raise ValueError("`search` parameter is required when `chat_id` is not provided.")
+
+        if chat_id:
+            session = self.session_manager.get_session(chat_id)
+            if session:
+                search = session.get("search")
+                generation = session.get("generation")
+                chat = session.get("chat")
+                request_timeout = session.get("request_timeout")
+                request_timeout_millis = session.get("request_timeout_millis")
+                request_options = session.get("request_options")
+                return self.chats.create_turns(
+                    chat_id=chat_id,
+                    query=query,
+                    search=search,
+                    generation=generation,
+                    chat=chat,
+                    request_options=request_options,
+                    request_timeout=request_timeout,
+                    request_timeout_millis=request_timeout_millis,
+                )
```

Comment:
If `chat_id` is not valid, it should return an error. otherwise the user might think that the `chat_id` is valid and keeps on sending it and a new session will be created

Reply:
I think we should send it to server even if we don't find chat id in the session because server would have that chat id and it can use the default config.

@ofermend what do you think?